cmplibrary/src/test/java/com/sourcepoint/cmplibrary/ConsentLibBuilderTest.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    @Test
+    public void testSetViewGroup() {
+        ViewGroup v = mock(ViewGroup.class);
+        consentLibBuilder.setViewGroup(v);
+        assertEquals(v, consentLibBuilder.viewGroup);
+    }
+
+    @Test
+    public void testSetOnMessageChoiceSelect() {
+        consentLibBuilder.setOnMessageChoiceSelect(callback);
+        assertEquals(callback, consentLibBuilder.onMessageChoiceSelect);
+    }
+
+    @Test
+    public void testSetOnInteractionComplete() {
+        consentLibBuilder.setOnInteractionComplete(callback);
+        assertEquals(callback, consentLibBuilder.onInteractionComplete);
+    }
+
+    @Test
+    public void testSetOnMessageReady() {
+        consentLibBuilder.setOnMessageReady(callback);
+        assertEquals(callback, consentLibBuilder.onMessageReady);
+    }
+
+    @Test
+    public void testSetOnErrorOccurred() {
+        consentLibBuilder.setOnErrorOccurred(callback);
+        assertEquals(callback, consentLibBuilder.onErrorOccurred);
+    }
+
+    @Test
+    public void testSetStage() {
+        boolean stage = true;
+        consentLibBuilder.setStage(stage);
+        assertEquals(stage, consentLibBuilder.stagingCampaign);
+    }
+
+    @Test
+    public void testSetInternalStage() {
+        boolean stage = true;
+        consentLibBuilder.setInternalStage(stage);
+        assertEquals(stage, consentLibBuilder.staging);
+    }
+
+    @Test
+    public void testSetInAppMessagePageUrl() {
+        String inAppMessageUrl = "inAppMessageUrl";
+        consentLibBuilder.setInAppMessagePageUrl(inAppMessageUrl);
+        assertEquals(inAppMessageUrl, consentLibBuilder.msgDomain);
+    }
+
+    @Test
+    public void testSetMmsDomain() {
+        String mmsDomain = "mmsDomain";
+        consentLibBuilder.setMmsDomain(mmsDomain);
+        assertEquals(mmsDomain, consentLibBuilder.mmsDomain);
+    }
+
+    @Test
+    public void testSetCmpDomain() {
+        String cmpDomain = "cmpDomain";
+        consentLibBuilder.setCmpDomain(cmpDomain);
+        assertEquals(cmpDomain, consentLibBuilder.cmpDomain);
+    }
+
+    @Test
+    public void testSetTargetingParamString() throws Exception {
+        String key = "key";
+        String stringValue = "stringValue";
+        EncodedParam targetingParamsString;
+        consentLibBuilder.setTargetingParam(key, stringValue);
+
+        Field localMember = getDeclaredFieldAccess("targetingParams");
+
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put(key, stringValue);
+
+        assertEquals(jsonObject.toString() , localMember.get(consentLibBuilder).toString());
+
+        targetingParamsString = new EncodedParam("targetingParams", jsonObject.toString());
+        Method method = ConsentLibBuilder.class.getDeclaredMethod("setTargetingParamsString");
+        method.setAccessible(true);
+        method.invoke(consentLibBuilder);
+        assertEquals(targetingParamsString.toString(), consentLibBuilder.targetingParamsString.toString());
+
+    }
```

Comment:
Let's keep one assertion per test case. So here I'd break this test case into two:

```java
@Test
public void setTargetingParamString() throws Exception {
  // test that the targeting param is indeed set
}

@Test
public void targetingParamStringIsEncoded() throws Exception {
  // test the value gets transformed into EncodedParam
}
```

We might still have some duplication this way, but then we have test cases which are very focused. If they break we'll know right away what's the assertion that broke.

Reply:
refactored the code to one assertion per method